sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClientOptions.cs

```diff
@@ -16,57 +16,127 @@ namespace Azure.Messaging.EventHubs.Producer
     ///
     internal class EventHubBufferedProducerClientOptions
     {
-        /// <summary> The number of batches that may be sent concurrently to each partition. </summary>
+        /// <summary>The number of batches that may be sent concurrently across all partitions.</summary>
+        private int _maximumConcurrentSends = Environment.ProcessorCount * 2;
+
+        /// <summary>The number of batches that may be sent concurrently to each partition.</summary>
         private int _maximumConcurrentSendsPerPartition = 1;
 
+        /// <summary>The total number of events that can be buffered for publishing at a given time for a given partition.</summary>
+        private int _maximumEventBufferLengthPerPartition = 1500;
+
+        /// <summary> The amount of time to wait for a new event to be enqueued in the buffer before publishing a partially full batch..</summary>
+        private TimeSpan? _maximumWaitTime = TimeSpan.FromMilliseconds(250);
+
         /// <summary>The set of options to use for configuring the connection to the Event Hubs service.</summary>
         private EventHubConnectionOptions _connectionOptions = new EventHubConnectionOptions();
 
         /// <summary>The set of options to govern retry behavior and try timeouts.</summary>
         private EventHubsRetryOptions _retryOptions = new EventHubsRetryOptions();
 
         /// <summary>
-        ///   The amount of time to wait for a new event to be added to the buffer before sending a partially
-        ///   full batch.
+        ///    Indicates whether or not events should be published using idempotent semantics for retries. If enabled, retries during publishing
+        ///    will attempt to avoid duplication with a minor cost to throughput.  Duplicates are still possible but the chance of them occurring is
+        ///    much lower when idempotent retries are enabled.
+        ///</summary>
+        ///
+        /// <value>
+        ///   By default, idempotent retries are disabled.
+        /// </value>
+        ///
+        /// <remarks>
+        ///   It is important to note that enabling idempotent retries does not guarantee exactly-once semantics.  The existing
+        ///   Event Hubs at-least-once delivery contract still applies and event duplication is unlikely, but possible.
+        /// </remarks>
+        ///
+        public bool EnableIdempotentRetries { get; set; }
```

Comment:
Is it semantically correct to call this idempotent? Based on the description is attempting to de-duplicate retries. I think of "exactly-once" and idempotent distinct concepts. My understanding of an idempotent operation is that it is resilient to any number of duplicate invocations by guaranteeing that the effect of a single operation is equivalent to N of the same operation.

Reply:
Discussed offline.  Conceptually and semantically, I'm in agreement.  However, the nomenclature is matched to a service feature and needs to remain in alignment.